### PR TITLE
feat(stages/webhook): Add Headers to Webhook stage https://github.com…

### PR DESCRIPTION
**Pull request by Nagarajj**
…/spinnaker/spinnaker/issues/1512

Add headers to webservice calls made as part of web hook stage
Also fix issue with "statusUrlResolution" not being initialized by default

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](http://www.spinnaker.io/docs/contributing-to-spinnaker).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

File: app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts [hunk 1]
```diff
@@ -29,12 +35,15 @@ export class WebhookStage {
...
-      statusUrlResolution: this.stage.statusUrlResolution || 'getMethod'
+      statusUrlResolution: this.stage.statusUrlResolution || 'getMethod',
+      headers: this.stage.headers || []
     };
 
     this.command = {
       payloadJSON: this.jsonUtilityService.makeSortedStringFromObject(this.stage.payload || {}),
     };
+
+    this.stage.statusUrlResolution = this.viewState.statusUrlResolution;
```
Comment (Nagarajj) [inline app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts]: Fix for a minor bug where statusUrlResolution is not initialized

**Comment by anotherchrisberry** [comment]
Seems to be basically a duplicate of https://github.com/spinnaker/deck/pull/3590 - @Nagarajj it looks like you're fixing the `statusUrlResolution` issue, which I don't think is happening in the other PR to add custom headers. Once we get the other PR merged, are you okay with issuing a separate PR to fix `statusUrlResolution`?

**Comment by Nagarajj** [comment]
@anotherchrisberry sure will do that. Thanks.

**Comment by anotherchrisberry** [comment]
closing in favor of https://github.com/spinnaker/deck/pull/3590 - looking forward to a new PR to fix the `statusUrlResolution` field

**Comment by Nagarajj** [comment]
@anotherchrisberry Have Submitted [PR](https://github.com/spinnaker/deck/pull/3601) to fix NPE, as statusUrlResolution is not set with default value of "getMethod" even though it is enabled in the UI by default.

**Pull request closed without merging**